### PR TITLE
Hermes 235

### PR DESCRIPTION
**Pull request by jmnarloch**
Adds awareness of trace information (like trace id).

### CODE REVIEW

**Comment by druminski** [comment]
:+1:

**Comment by adamdubiel** [comment]
:+1:

File: hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/di/FrontendBinder.java [hunk 1]
```diff
@@ -36,6 +38,7 @@ protected void configure() {
 
         bindSingleton(SchemaPrefetchService.class);
         bindSingleton(HealthCheckService.class);
+        bind(HeaderTraceIdExtractor.class).to(TraceExtractor.class).in(Singleton.class);
```
Comment (adamdubiel) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/di/FrontendBinder.java]: could we expose this in `FrontendBuilder` and `ConsumersBuilder`? now you need to know about this secret setting

Comment (jmnarloch) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/di/FrontendBinder.java]: Ok, both from what I can understand you can provide your own binding directly through the builder: https://github.com/allegro/hermes/blob/master/hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/HermesFrontend.java#L159

So is there a need to explicitly provide method for that?

**Pull request merged**